Airflow documentation updates for Windows WSL2 installation (#57027)
mistercrunch/Airflow: airflow-core/docs/start.rst

Reviewer: ```suggestion
   Officially supported installation methods is with ``pip`` or ``uv``.
```

Reviewer: ```suggestion
   Run ``pip install apache-airflow[EXTRAS]==AIRFLOW_VERSION --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-AIRFLOW_VERSION/constraints-PYTHON_VERSION.txt"``, for example ``pip install "apache-airflow[celery]==3.0.0" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-3.0.0/constraints-3.10.txt"`` to install Airflow in a reproducible way. You can also use - much faster -  ``uv`` - by just adding ``uv`` before the command.
```

   This guide will help you quickly set up Apache Airflow using ``uv``, a fast and modern tool for managing Python environments and dependencies. ``uv`` makes the installation process easy and provides a
   smooth setup experience.

   If you are on Windows, you have to use WSL2 (Linux environment for Windows).

>       .. code-block bash

Reviewer: This is now formatted as comment in .rst (it will not be visible in the rendered page).

Author: Thanks for your inputs! Incorporated changes below.

         wsl --install
         # Restart computer, then in WSL Ubuntu terminal
         sudo apt update
         sudo apt install python3-pip python3-venv

         bash # Go to Linux home directory (not Windows mount)
         cd ~

         # Create airflow directory
         mkdir -p ~airflow
         cd ~airflow

>          # Create virtual environment

Reviewer: As discussed before -> the creation of the environment is the same in WSL2 and outside - so we should rather update point 2. below`Install airflow using uv` to be `Install airflow in a virtual environment` - and rather add explanation that `uv` is a faster alternative to `pip` and creates the venv automatically for you - but both options - with `uv` and withoug (just with venv + pip) should be explained there.

Author: Thanks for your inputs! Incorporated changes below.

      uv pip install "apache-airflow==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"

> 4. Install Airflow on Windows, using WSL2. In Powershell, run the following commands to create WSL2 environment.

Reviewer: ```suggestion
If you are on Windows, you have to use WSL2 (Linux environment for Windows). 

In Powershell, run the following commands to create WSL2 environment and install Airflow:
```

Reviewer: Actually - this whole chapter should be way above - before numboer 1. - as optional prerequiste if you are on Windows (and it should only be about installing WSL2 - all the rest should be the same, so there is no need to repeat it.

That will make the whole "start.rst" way shorter (which should be the most important goal for that document).

   .. code-block:: bash

      :substitutions:

      wsl --install
      # Restart computer, then in WSL Ubuntu terminal:
      sudo apt update
      sudo apt install python3-pip python3-venv

      bash # Go to Linux home directory (not Windows mount)
      cd ~

      # Create airflow directory
      mkdir -p ~/airflow
      cd ~/airflow

      # Create virtual environment
      python3 -m venv airflow_venv

      # Activate
      source airflow_venv/bin/activate

      # Upgrade pip
      pip install --upgrade pip

      # Install Airflow with correct Python version constraints
      pip install "apache-airflow[celery]==3.1.0" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-3.1.0/constraints-3.12.txt"

      # Verify installation
      airflow version

> 5. Run Airflow Standalone:

Reviewer: ```suggestion
4. Run Airflow Standalone:
```

   The ``airflow standalone`` command initializes the database, creates a user, and starts all components.

   .. code-block:: bash

      airflow standalone

   For accessing admin account details for Airflow installation on Windows WSL2, use contents in file ``simple_auth_manager_passwords.json.generated`` to login to Airflow UI.

> 6. Access the Airflow UI:

Reviewer: ```suggestion

5. Access the Airflow UI:
```

Reviewer: There is no difference WSL2 vs Mac/Linux
